.github/workflows/sizeComparison.yml

```diff
@@ -0,0 +1,66 @@
+name: sizeComparison
+
+on:
+  push:
+    branches:
+      - main*
+      - fluent2*
+  pull_request:
+    branches:
+      - main*
+      - fluent2*
+
+jobs:
+  archiveHeadBuild:
+    runs-on: macos-12
+    strategy:
+      fail-fast: false
+    steps: 
+    - uses: actions/checkout@v3 # Checkout repo at head sha
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+    - name: Switch to current version of Xcode
+      run: scripts/xcode_select_current_version.sh
+    - name: scripts/xcodebuild_wrapper.sh 'ios_device_build Demo.Development Release build -archivePath $GITHUB_WORKSPACE/headBuild archive' # Build and archive release device build
+      run: scripts/xcodebuild_wrapper.sh ios_device_build Demo.Development Release build -archivePath $GITHUB_WORKSPACE/${{ github.event.pull_request.head.sha }} archive
+    - name: Create size file # Get size of demo app in "kilobytes" (2*512B blocks) and store in a file with the hash as the name
+      run: du -sk $GITHUB_WORKSPACE/${{ github.event.pull_request.head.sha }}.xcarchive/Products/Applications/FluentUI.Demo.app | cut -f1 >> $GITHUB_WORKSPACE/${{ github.event.pull_request.head.sha }}
```

Comment:
is the `pull-request` object populated on 'push' jobs as well?

Reply:
GitHub's docs don't really have a 'reference guide'  section. it all just examples and general terms....

Reply:
I believe so. It seemed to be properly updating when I was testing it.

Reply:
Ah. I misunderstood the question. I'm not actually sure if it gets populated on just a push to a branch, and not during a PR. I can check.